### PR TITLE
KE-28134-02

### PR DESCRIPTION
**Pull request by songzhxlh-max**
## What changes were proposed in this pull request?
fix bug for KE-28134 of "outputPartition column effective scope only inner of the sortMergeJoin and switch"

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala [hunk 1]
```diff
@@ -285,10 +328,31 @@ case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
 
 object EnsureRequirements {
 
+  val isHashPartitioningFromCube = "CUBE_HASH_PARTITIONING"
+
+  val isSortMergeJoinExec = "SortMergeJoinExec"
```
Comment (fishcus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala:333]: isHashPartitioningFromCube rename
isSortMergeJoinExec这个用到了？

Comment (songzhxlh-max) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala:333]: 用了，241行 && child.getClass.getCanonicalName.contains(EnsureRequirements.isSortMergeJoinExec)) {

Comment (fishcus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala:333]: isHashPartitioningFromCube rename
isSortMergeJoinExec这个用到了？

File: sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala [hunk 2]
```diff
@@ -285,10 +331,28 @@ case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
 
 object EnsureRequirements {
 
+  val CubeHashPartitioning = "CUBE_HASH_PARTITIONING"
```
Comment (fishcus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala:334]: 常量的话  大写 _ 分割

**Pull request closed without merging**